samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java

```diff
@@ -348,24 +356,30 @@ private boolean isErrorTransient(Throwable throwable) {
     return new LinkedBlockingQueue<>(config.getConsumerBufferCapacity(systemName));
   }
 
-  protected class PartitionReceiverHandlerImpl extends PartitionReceiveHandler {
+  protected class PartitionReceiverHandlerImpl implements PartitionReceiveHandler {
 
     private final Counter eventReadRate;
     private final Counter eventByteReadRate;
     private final SamzaHistogram readLatency;
     private final Counter errorRate;
     private final Interceptor interceptor;
+    private final Integer maxEventCount;
     SystemStreamPartition ssp;
 
     PartitionReceiverHandlerImpl(SystemStreamPartition ssp, Counter eventReadRate, Counter eventByteReadRate,
-                                 SamzaHistogram readLatency, Counter readErrors, Interceptor interceptor) {
-      super(MAX_EVENT_COUNT_PER_PARTITION_POLL);
+        SamzaHistogram readLatency, Counter readErrors, Interceptor interceptor) {
       this.ssp = ssp;
       this.eventReadRate = eventReadRate;
       this.eventByteReadRate = eventByteReadRate;
       this.readLatency = readLatency;
       this.errorRate = readErrors;
       this.interceptor = interceptor;
+      this.maxEventCount = config.getMaxEventCountPerPoll(systemName);
```

Comment:
Minor: Prefer passing as an explicit constructor param. For consistency, as well as avoiding access to parent class state.